Makefile seems to have deprecated k3d commands

## Summary 

What happened/what you expected to happen?

After starting k3d cluster locally, if I run `make start` I get an error
```
➜  argo git:(learning) ✗ make start
# GIT_TAG=v2.10.0-rc1, GIT_BRANCH=learning, GIT_TREE_STATE=dirty, VERSION=latest, DEV_IMAGE=true, K3D=true
killall argo workflow-controller kubectl || true
No matching processes belonging to you were found
k3d start
Error: unknown command "start" for "k3d"
Run 'k3d --help' for usage.
FATA[0000] unknown command "start" for "k3d"            
make: *** [install] Error 1
➜  argo git:(learning) ✗ kubectl cluster-info
Kubernetes master is running at https://0.0.0.0:61662
CoreDNS is running at https://0.0.0.0:61662/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
Metrics-server is running at https://0.0.0.0:61662/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

## Diagnostics

What Kubernetes provider are you using? 
`K3d`
What version of Argo Workflows are you running? 
```
argo: latest+9319c07.dirty
  BuildDate: 2020-10-03T06:45:57Z
  GitCommit: 9319c074e742c5d9cb97d6c5bbbf076afe886f76
  GitTreeState: dirty
  GitTag: v2.10.0-rc1
  GoVersion: go1.14.3
  Compiler: gc
  Platform: darwin/amd64
```
This line here for ref: https://github.com/argoproj/argo/blob/300634af42a3017a5354468a2779ef1fa5e37931/Makefile#L326
```yaml
Paste a workflow that reproduces the bug, including status:
kubectl get wf -o yaml ${workflow} 
```

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/.github/workflows/ci-build.yaml
+++ b/.github/workflows/ci-build.yaml
@@ -2,12 +2,12 @@ name: CI
 on:
   push:
     branches:
-      - 'master'
-      - 'release-*'
-      - '!release-2.8'
+      - "master"
+      - "release-*"
+      - "!release-2.8"
   pull_request:
     branches:
-      - 'master'
+      - "master"
 
 jobs:
   tests:
@@ -29,14 +29,14 @@ jobs:
       - name: Setup Golang
         uses: actions/setup-go@v1
         with:
-          go-version: '1.13.12'
+          go-version: "1.13.12"
       - name: Add bins to PATH
         run: |
           echo "::add-path::/home/runner/go/bin"
           echo "::add-path::/usr/local/bin"
-      - name: Install and start K3S v1.0.1
+      - name: Install and start K3S v1.18.8+k3s1
         if: ${{ matrix.test != 'test' }}
-        run: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.0.1 INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh - &
+        run: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.18.8+k3s1 INSTALL_K3S_CHANNEL=stable INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh - &
       - name: Pre-pull images
         if: ${{ matrix.test != 'test' }}
         env:
@@ -97,7 +97,7 @@ jobs:
       - name: Setup Golang
         uses: actions/setup-go@v1
         with:
-          go-version: '1.13.12'
+          go-version: "1.13.12"
       - name: Add bins to PATH
         run: |
           echo "::add-path::/home/runner/go/bin"
@@ -140,7 +140,7 @@ jobs:
         uses: actions/checkout@v2
       - uses: actions/setup-node@v1
         with:
-          node-version: '14'
+          node-version: "14"
       - name: Restore node dependency cache
         uses: actions/cache@v1
         with:

--- a/Makefile
+++ b/Makefile
@@ -62,7 +62,7 @@ endif
 # * `local` run the workflow–controller and argo-server as single replicas on the local machine (default)
 # * `kubernetes` run the workflow-controller and argo-server on the Kubernetes cluster
 RUN_MODE              := local
-K3D                   := $(shell if [ "`which kubectl`" != '' ] && [ "`kubectl config current-context`" = "k3s-default" ]; then echo true; else echo false; fi)
+K3D                   := $(shell if [[ "`which kubectl`" != '' ]] && [[ "`kubectl config current-context`" == "k3d-"* ]]; then echo true; else echo false; fi)
 LOG_LEVEL             := debug
 UPPERIO_DB_DEBUG      := 0
 NAMESPACED            := true
@@ -112,12 +112,12 @@ define docker_build
 	if [ $(DEV_IMAGE) = true ]; then $(MAKE) dist/$(2)-$(OUTPUT_IMAGE_OS)-$(OUTPUT_IMAGE_ARCH) && mv dist/$(2)-$(OUTPUT_IMAGE_OS)-$(OUTPUT_IMAGE_ARCH) $(2); fi
 	docker build --progress plain -t $(IMAGE_NAMESPACE)/$(1):$(VERSION) --target $(1) -f $(DOCKERFILE) --build-arg IMAGE_OS=$(OUTPUT_IMAGE_OS) --build-arg IMAGE_ARCH=$(OUTPUT_IMAGE_ARCH) .
 	if [ $(DEV_IMAGE) = true ]; then mv $(2) dist/$(2)-$(OUTPUT_IMAGE_OS)-$(OUTPUT_IMAGE_ARCH); fi
-	if [ $(K3D) = true ]; then k3d import-images $(IMAGE_NAMESPACE)/$(1):$(VERSION); fi
+	if [ $(K3D) = true ]; then k3d image import $(IMAGE_NAMESPACE)/$(1):$(VERSION); fi
 	touch $(3)
 endef
 define docker_pull
 	docker pull $(1)
-	if [ $(K3D) = true ]; then k3d import-images $(1); fi
+	if [ $(K3D) = true ]; then k3d image import $(1); fi
 endef
 
 .PHONY: build
@@ -342,9 +342,6 @@ dist/$(PROFILE).yaml: $(MANIFESTS) $(E2E_MANIFESTS) /usr/local/bin/kustomize
 
 .PHONY: install
 install: dist/$(PROFILE).yaml
-ifeq ($(K3D),true)
-	k3d start
-endif
 	cat test/e2e/manifests/argo-ns.yaml | sed 's/argo/$(KUBE_NAMESPACE)/' > dist/argo-ns.yaml
 	kubectl apply -f dist/argo-ns.yaml
 	kubectl -n $(KUBE_NAMESPACE) apply -l app.kubernetes.io/part-of=argo --prune --force -f dist/$(PROFILE).yaml
@@ -357,7 +354,7 @@ pull-build-images:
 argosay: test/e2e/images/argosay/v2/argosay
 	cd test/e2e/images/argosay/v2 && docker build . -t argoproj/argosay:v2
 ifeq ($(K3D),true)
-	k3d import-images argoproj/argosay:v2
+	k3d image import argoproj/argosay:v2
 endif
 	docker push argoproj/argosay:v2
 

--- a/USERS.md
+++ b/USERS.md
@@ -1,10 +1,13 @@
 ## Argo Workflows User Community Surveys & Feedback
+
 Please find [here](community/ArgoWorkflows2020SurveySummary.pdf) Argo Workflows user community 2020 survey results.
 
 ## Who uses Argo Workflows?
+
 As the Argo Community grows, we'd like to keep track of our users. Please send a PR with your organization name.
 
 Currently, the following organizations are **officially** using Argo Workflows:
+
 1. [23mofang](https://www.23mofang.com/)
 1. [Adevinta](https://www.adevinta.com/)
 1. [Admiralty](https://admiralty.io/)
@@ -90,3 +93,4 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [Wavefront](https://www.wavefront.com/)
 1. [Wellcome Trust](https://wellcome.ac.uk/)
 1. [Helio](https://helio.exchange)
+1. [HSBC](https://hsbc.com)

